File: client/src/main/java/commands/pcommands/ExecuteScript.java
package commands.pcommands;

import commands.*;
import exceptions.ScriptExecutionException;
import utils.ScriptExecutor;

public final class ExecuteScript extends AbstractCommand {
  private final ScriptExecutor executor;

  public ExecuteScript(ScriptExecutor executor) {
    super(CommandInfo.of("execute_script", "Run all commands from specified file", true, 1, false, ExecutionMode.CLIENT));
    this.executor = executor;
  }
  
  @Override
  public ExecutionResult execute(ExecutionPayload payload) {
    String fileName = payload.getInlineArg();
    try {
      executor.executeScript(fileName);
    } catch (ScriptExecutionException e) {
      return ExecutionResult.valueOf(false, e.getMessage());
    }
    return ExecutionResult.valueOf(true, "Script execution started");
  }
}

File: client-ui/domain/src/commonMain/kotlin/ru.moleus.kollector/domain/commands/pcommands/ExecuteScript.kt
package ru.moleus.kollector.domain.commands.pcommands

import commands.*
import ru.moleus.kollector.domain.client.ScriptExecutor
import ru.moleus.kollector.domain.exceptions.ScriptExecutionException

class ExecuteScript(executor: ScriptExecutor) : AbstractCommand(
    CommandInfo.of(
        "execute_script",
        "Run all commands from specified file",
        true,
        1,
        CommandType.OTHER,
        ExecutionMode.CLIENT
    )
) {
    private val executor: ScriptExecutor

    init {
        this.executor = executor
    }

    override fun execute(payload: ExecutionPayload): ExecutionResult {
        val fileName: String = payload.inlineArg
        try {
            executor.executeScript(fileName)
        } catch (e: ScriptExecutionException) {
            return ExecutionResult.valueOf(false, e.message)
        }
        return ExecutionResult.valueOf(true, "Script execution started")
    }
}